### PR TITLE
FALCON-1908 Document Safemode in Falcon Server

### PR DESCRIPTION
**Pull request by bvellanki**

### CODE REVIEW

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 1]
```diff
@@ -91,6 +91,47 @@ Startup property file in both falcon & prism server need to be configured with t
 
 ---++++ Prism Setup
 <img src="PrismSetup.png" height="400" width="600" />
+
+---+++ Safe Mode
+Safemode is useful when the admin wants to prevent Falcon users from scheduling entities in the workflow engine. This can happen when
+   * Hadoop clusters are being upgraded.
+   * Falcon cluster entity is being updated.
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: Better to use plurality to be consistent with the previous sentence, i.e. "Falcon cluster entities are being updated".

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 2]
```diff
@@ -91,6 +91,47 @@ Startup property file in both falcon & prism server need to be configured with t
...
+
+When in Safemode, Falcon Server restricts users from performing certain operations. The restrictions are as follows.
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: You talked about what is allowed and then restrictions. How about: "In safemode, users can only perform limited operations. To be specific,"

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 3]
```diff
@@ -91,6 +91,47 @@ Startup property file in both falcon & prism server need to be configured with t
...
+   * Suspend/Kill of individual instances will be allowed if users want to suspend specific job operations.
+   * Suspend operation on schedulable entities will be allowed. This is because, a user might want to suspend entities during rolling-upgrade to handle jobs incompatible with updated versions.
+   * All other operations are not allowed. To enumerate,
+      * All entity submit, submitAndSchedule operations are not allowed.
+      * Entity operations not allowed are : update, schedule, touch, delete, submit, submitAndSchedule, resume
+      * Instance operations not allowed are : rerun, resume
+
+---++++ Getting into/out-of Safemode.
+Falcon process owner can start Falcon in safemode by using following command. When -setsafemode is true, Falcon starts in safemode.
+When -setsafemode is false, falcon starts in normal mode.
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: How about: "The user can specify whether to start Falcon in safemode with the following command:"?

**Comment by bvellanki** [comment]
@yzheng-hortonworks  Good feedback, I incorporated it into the second commit. Please review.

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 4]
```diff
@@ -91,6 +91,46 @@ Startup property file in both falcon & prism server need to be configured with t
...
+   * Hadoop clusters are being upgraded.
+   * Falcon cluster entities are being updated.
+
+When in Safemode, users can only perform limited operations. To be specific,
+   * Users can perform read operations on entities subject to authorization.
+   * Superuser can perform cluster entity update operation.
+   * Suspend/Kill of individual instances will be allowed if users want to suspend specific job operations.
+   * Suspend operation on schedulable entities will be allowed. This is because, a user might want to suspend entities during rolling-upgrade to handle jobs incompatible with updated versions.
+   * All other operations are not allowed. To enumerate,
+      * All entity submit, submitAndSchedule operations are not allowed.
```
Comment (vramachan) [inline docs/src/site/twiki/FalconDocumentation.twiki]: redundant?

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 5]
```diff
@@ -91,6 +91,46 @@ Startup property file in both falcon & prism server need to be configured with t
...
+<falcon-server>/bin/falcon admin  [-setsafemode <true/false>]
+
+## RestAPI
+GET http://falcon-server:15000/api/admin/setSafeMode/true
+OR
+GET http://falcon-server:15000/api/admin/setSafeMode/false
+</verbatim>
+
+User can find if FalconServer is in safemode or not, by calling the [[restapi/AdminVersion][Admin Version]] API. Once
+server is set to safemode, this state is persisted during restarts. It has to be unset explicitly if user wants to exit safemode.
```
Comment (vramachan) [inline docs/src/site/twiki/FalconDocumentation.twiki]: can we please add a NOTE before this in bold.

**Comment by yzheng-hortonworks** [comment]
+1 after addressing @vramachan 's comments

**Comment by vramachan** [comment]
some minor comments. 
+1

File: docs/src/site/twiki/FalconDocumentation.twiki (hunk 4 above)
Comment (bvellanki) [inline docs/src/site/twiki/FalconDocumentation.twiki]: No, I think it is better to spell out specifically what is allowed and what is not.

**Pull request closed without merging**